src/main/java/bio/terra/service/JobService.java

```diff
@@ -179,7 +164,25 @@ public JobModel mapFlightStateToJobModel(FlightState flightState) {
 
     public List<JobModel> enumerateJobs(
         int offset, int limit, AuthenticatedUserRequest userReq) {
-        List<FlightState> flightStateList = stairway.getFlightsForUser(offset, limit, buildUserRequestInfo(userReq));
+        boolean canListAnyJob = true;
+        if (userReq != null) {
+            // currently, this check will be true for stewards only
+            canListAnyJob = samService.isAuthorized(
+                userReq,
+                SamClientService.ResourceType.DATAREPO,
+                appConfig.getResourceId(),
+                SamClientService.DataRepoAction.LIST_JOBS);
+        }
```

Comment:
I think we should throw if `userReq` is null.

Reply:
Same comment as above.

Reply:
Again, same comment as above.